Makefile

```diff
@@ -795,11 +795,10 @@ pkgs: build-tools $(PKGS)
 
 pkg/external-boot-image/build.yml: pkg/external-boot-image/build.yml.in
 	$(QUIET)tools/compose-external-boot-image-yml.sh $< $@ $(shell echo ${KERNEL_TAG} | cut -d':' -f2) $(shell $(LINUXKIT) pkg show-tag pkg/xen-tools | cut -d':' -f2)
-pkg/external-boot-image: pkg/external-boot-image/build.yml
-pkg/kube: pkg/external-boot-image
-	$(MAKE) eve-external-boot-image
-	$(MAKE) cache-export IMAGE=$(shell $(LINUXKIT) pkg show-tag pkg/external-boot-image | tee pkg/kube/external-boot-image.tag) OUTFILE=pkg/kube/external-boot-image.tar && \
-	$(MAKE) eve-kube && rm -f pkg/kube/external-boot-image.tar pkg/kube/external-boot-image.tag pkg/external-boot-image/build.yml
+eve-external-boot-image: pkg/external-boot-image/build.yml
+pkg/kube/external-boot-image.tar: pkg/external-boot-image
+	$(MAKE) cache-export IMAGE=$(shell $(LINUXKIT) pkg show-tag pkg/external-boot-image) OUTFILE=pkg/kube/external-boot-image.tar
+pkg/kube: pkg/kube/external-boot-image.tar eve-kube
```

Comment:
I don't understand this line. Isn't `eve-kube` already a dependency on (shorthand for) `pkg/kube`, but the rule cited in [this line](https://github.com/lf-edge/eve/blob/9011acdf7d3822fe67841d7871cad21bdc4f1a80/Makefile#L1013)? I get the first part:

```Makefile
pkg/kube: pkg/kube/external-boot-image.tar
```

but the addition of `eve-kube` confuses me. What is it there for? Isn't that circular?

Reply:
the pkg/kube target follows the same pattern as the pkg/pillar target below it:

`pkg/pillar: pkg/dnsmasq pkg/gpt-tools pkg/dom0-ztools eve-pillar`

makefile target dependency ordering follows left to right so pkg/kube will process them in order:
- pkg/kube/external-boot-image.tar
- eve-kube

Reply:
I didn't realize it does. That is strange. But ok, if it works, it works.